pq-src/build.rs

```diff
@@ -181,90 +183,93 @@ int main() {"#,
 make_test_for!(TEST_FOR_STRCHRNUL, r#"strchrnul("", 42);"#);
 make_test_for!(TEST_FOR_STRSIGNAL, r#"strsignal(32);"#);
 
-fn check_compiles(test: &str) -> bool {
-    let test_path = std::env::var("OUT_DIR").expect("Set by cargo") + "/test.c";
-    std::fs::write(&test_path, &test).expect("Failed to write test");
-    let r = cc::Build::new().file(&test_path).try_compile("test");
-    std::fs::remove_file(test_path).expect("Failed to remove test file");
+fn check_compiles(test: &str, mut command: cc::Build) -> bool {
+    // Add necessary compiler-flags to make sure that undefined functions actually cause errors
+    // NOTE: Starting with GCC 14 and Clang 16, this is default behaviour.
+    // See:
+    //   - https://gcc.gnu.org/gcc-14/porting_to.html#implicit-function-declaration
+    //   - https://releases.llvm.org/16.0.0/tools/clang/docs/ReleaseNotes.html#potentially-breaking-changes
+    command
+        .flag_if_supported("-Werror=implicit-function-declaration") // GCC/Clang
+        .flag_if_supported("/we4013"); // MSVC
```

Comment:
One potential issue: If a compiler other than GCC/Clang/MSVC is used, this might not add either of these flags and implicit declarations might still silently work.
However I think this is a non-issue, as those three likely represent 99,9% of all compilation that is done.